rskj-core/src/main/java/org/ethereum/db/TrieKeyMapper.java

```diff
@@ -45,8 +45,9 @@
             return Arrays.copyOf(key, key.length);
         }
 
-        byte[] secureKey = secureKeyPrefix(addr.getBytes());
-        byte[] key = ByteUtil.merge(DOMAIN_PREFIX, secureKey, addr.getBytes());
+        byte[] key = mapRskAddressToKey(addr);
+
+        accountKeys.put(addr, key);
```

Comment:
I would add a comment explaining that this cache is safe because we a RskAddress is immutable and it's a purely functional mapping

Reply:
done